src/aura-fed/BalancerAdapter.sol

```diff
@@ -48,58 +48,58 @@ contract BalancerComposableStablepoolAdapter {
-        swapExactIn(address(dola), address(bpt), dolaAmount, minBptOut);
-        uint bptOut =  bpt.balanceOf(address(this)) - init;
-        return bptOut;
+    function _addLiquidity(uint dolaAmount, uint maxSlippage) internal returns(uint){
+        uint initialBal = BPT.balanceOf(address(this));
+        uint BPTWanted = bptNeededForDola(dolaAmount);
+        uint minBptOut = BPTWanted - BPTWanted * maxSlippage / BPS;
+        swapExactIn(address(_DOLA), address(BPT), dolaAmount, minBptOut);
+        uint BPTOut =  BPT.balanceOf(address(this)) - initialBal;
+        return BPTOut;
     }
     
     /**
     @notice Withdraws an amount of value close to dolaAmount
     @dev Will rarely withdraw an amount equal to dolaAmount, due to slippage.
-    @param dolaAmount Amount of dola the withdrawer wants to withdraw
-    @param maxSlippage Maximum amount of value that can be lost in basis points, assuming DOLA = 1$
+    @param dolaAmount Amount of _DOLA the withdrawer wants to withdraw
+    @param maxSlippage Maximum amount of value that can be lost in basis points, assuming _DOLA = 1$
     */
-    function _withdraw(uint dolaAmount, uint maxSlippage) internal returns(uint){
-        uint init = dola.balanceOf(address(this));
-        uint bptNeeded = bptNeededForDola(dolaAmount);
+    function _removeLiquidity(uint dolaAmount, uint maxSlippage) internal returns(uint){
+        uint initialBal = _DOLA.balanceOf(address(this));
+        uint BPTNeeded = bptNeededForDola(dolaAmount);
         uint minDolaOut = dolaAmount - dolaAmount * maxSlippage / BPS;
-        swapExactIn(address(bpt), address(dola), bptNeeded, minDolaOut);
-        uint dolaOut = dola.balanceOf(address(this)) - init;
+        swapExactIn(address(BPT), address(_DOLA), BPTNeeded, minDolaOut);
+        uint dolaOut = _DOLA.balanceOf(address(this)) - initialBal;
         return dolaOut;
     }
 
     /**
     @notice Withdraws all BPT in the contract
     @dev Will rarely withdraw an amount equal to dolaAmount, due to slippage.
-    @param maxSlippage Maximum amount of value that can be lost in basis points, assuming DOLA = 1$
+    @param maxSlippage Maximum amount of value that can be lost in basis points, assuming _DOLA = 1$
     */
-    function _withdrawAll(uint maxSlippage) internal returns(uint){
-        uint bptBal = bpt.balanceOf(address(this));
-        uint expectedDolaOut = bptBal * bpt.getRate() / 10**18;
+    function _removeAllLiquidity(uint maxSlippage) internal returns(uint){
+        uint BPTBal = BPT.balanceOf(address(this));
+        uint expectedDolaOut = BPTBal * BPT.getRate() / 10**18;
         uint minDolaOut = expectedDolaOut - expectedDolaOut * maxSlippage / BPS;
-        swapExactIn(address(bpt), address(dola), bptBal, minDolaOut);
-        return dola.balanceOf(address(this));
+        swapExactIn(address(BPT), address(_DOLA), BPTBal, minDolaOut);
+        return _DOLA.balanceOf(address(this));
```

Comment:
info:
Inconsistent return value behavior between `_removeLiquidity()` and `_removeAllLiquidity()`.
The latter does not account for any pre-exiting DOLA in the contract.

Reply:
It's working as intended. When calling the `_removeLiquidity` function, the intention for the Fed should be to remove a certain amount of DOLA from the active market, where `_removeAllLiquidity` is meant to wind down the Fed operation.
I suppose we could return the additional DOLA, but that's ultimately immaterial, as aslong the DOLA is outside of the market, then it's not influencing prices, and we can either take it as profit or burn it when winding down the Fed. In the general, DOLA existing within the mainnet AuraFed would be an anomaly.